### PR TITLE
Ignore falsy values in proto encoding

### PR DESCRIPTION
**Pull request by RodrigoAD**

### CODE REVIEW

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/setup.dev.flow.ts [hunk 1]
```diff
@@ -51,9 +51,9 @@ export default class SetupFlow extends FlowCommand<TransactionResponse> {
         '9n1sSGA5rhfsQyaX3tHz3ZU1ffR6V8KffvWtFPBcFrJw',
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/setup.dev.flow.ts:51]: I see this file is used as a means of debugging and we plan to remove it?
If it's useful, is there an option to restructure this as a test?

File: gauntlet/packages/gauntlet-solana-contracts/src/core/proto/index.ts [hunk 2]
```diff
@@ -22,14 +22,30 @@ export class Protobuf {
...
+    return copy
+  }
+
   encode = (type: string, payload: any): Uint8Array => {
     const protoType = this.root.lookupType(type)
-    let err = protoType.verify(payload)
+    // Proto encoding should ignore falsy values. In protobuf.js we need to assign them to to be ignored
+    // https://docs.cosmos.network/master/architecture/adr-027-deterministic-protobuf-serialization.html#implementation
+    const noFalsyPayload = this.falsyToNullProperties(payload)
+    let err = protoType.verify(noFalsyPayload)
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-solana-contracts/src/core/proto/index.ts:43]: This error object potentially bears some valuable info.

I'd suggest throwing that `err` object so it's surfaced, and just log the "`Protobuf: Payload does not match descriptor ${type}`" message.

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/setup.dev.flow.ts (hunk 1 above)
Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/setup.dev.flow.ts:51]: That's exactly the plan, move it to a test once we have the framework

**Comment by RodrigoAD** [comment]
This should be ready @krebernisak

**Review by krebernisak — APPROVED** [review]

**Pull request merged**